src/north/CsvToHttp/CsvToHttp.class.js

```diff
@@ -0,0 +1,230 @@
+        },
+        step: (step) => {
+          csvDataParsed.push(step.data)
+        },
+        complete: () => {
+          // The csv parsing is a success so we begin to map it
+          // Initialize the body to an empty array
+          this.httpBody = []
+          this.convertToHttpBody(csvDataParsed)
+        },
+      })
+    } catch (error) {
+      this.logger.error(error)
+    }
+  }
+
+  /**
+   * @param {Object} jsonObject - jsonObj
+   * @return {Promise} - The send status
+   */
+  convertToHttpBody(jsonObject) {
+    try {
+      if (jsonObject.length > 0) {
+        jsonObject.forEach((jsonObj) => {
+          const response = this.convertCSVRowIntoHttpBody(jsonObj)
+          // Test the result of the mapping/convertion before add it in the httpBody
+          if (response) {
+            this.httpBody.push(response)
+          }
+        })
+      }
+    } catch (error) {
+      this.logger.error(error)
+      throw ApiHandler.STATUS.COMMUNICATION_ERROR
+    }
+  }
+
+  /**
+   * @param {Object} jsonObject - jsonObj
+   * @return {Object} - The send status
+   */
+  convertCSVRowIntoHttpBody(jsonObject) {
+    const object = {}
+    this.mapping.forEach((mapping) => {
+      if (jsonObject[mapping.csvField]) {
+        const field = mapping.httpField
+        const value = this.convertToCorrectType(jsonObject[mapping.csvField], mapping.type)
+
+        object[field] = value
+      }
+    })
+    return object
+  }
+
+  /**
+   * Convert the value in the selected type (string by default)
+   * @param {mixed} value - value
+   * @param {string} type - type
+   * @return {mixed} - The converted value
+   */
+  convertToCorrectType(value, type) {
+    let valueConverted
+    switch (type) {
+      case 'integer':
+        try {
+          valueConverted = parseInt(value)
+        } catch (e) {
+          this.logger.error(e)
+          valueConverted = value
+        }
+        break
+      case 'float':
+        try {
+          valueConverted = parseFloat(value)
+        } catch (e) {
+          this.logger.error(e)
+          valueConverted = value
+        }
+        break
+      case 'timestamp':
+        try {
+          valueConverted = parseInt(value)
+        } catch (e) {
+          this.logger.error(e)
+          valueConverted = value
+        }
+        break
+      case 'date (ISO)':
+        try {
+          valueConverted = new Date(value)
+        } catch (e) {
+          this.logger.error(e)
+          valueConverted = value
+        }
+        break
+      default:
+        return value
+    }
+    return valueConverted
+  }
+
+  /**
+   * @return {Promise} - Promise
+   */
+  async sendData() {
+    return new Promise((resolve, reject) => {
+      try {
+        const maxLength = 100
+
+        if (this.httpBody.length > maxLength) {
+          // Divide the current body in array of maximun maxLength elements
+          let i = 0
+          while (i < this.httpBody.length) {
```

Comment:
OK, this one is not your fault but my phobia for "while" loops :-) 
look at https://www.iditect.com/how-to/50650976.html to find inspiration